### PR TITLE
fix(receive): Adds single or multiple modal for receive

### PR DESCRIPTION
**Pull request by jorbuedo**
related: https://emurgo.atlassian.net/browse/YOMO-1258

image

### CODE REVIEW

**Review by michalszorad — APPROVED** [review]

**Review by SorinC6 — APPROVED** [review]

File: apps/wallet-mobile/src/features/Receive/useCases/DescribeSelectedAddressScreen.tsx [hunk 1]
```diff
@@ -31,6 +39,10 @@ export const DescribeSelectedAddressScreen = () => {
     copy(selectedAddress)
   }
 
+  React.useEffect(() => {
+    isSingle && isShowingMultipleAddressInfo && openModal(strings.singleOrMultiple, <Modal />, modalHeight)
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Receive/useCases/DescribeSelectedAddressScreen.tsx]: hey @jorbuedo by reading the ticket it looks like it should check for the used addresses and if there is more than 1  suggest the change, isSingle here is the current config only and the isShowing is for the popup, I think it needs to check for the wallet.usedAddress... to decide if whether it should open or not, can you check with Product?

Comment (jorbuedo) [reply apps/wallet-mobile/src/features/Receive/useCases/DescribeSelectedAddressScreen.tsx]: Yes, only needs to show when there's multiple used already

File: apps/wallet-mobile/src/features/Receive/useCases/DescribeSelectedAddressScreen.tsx [hunk 2]
```diff
@@ -31,6 +42,13 @@ export const DescribeSelectedAddressScreen = () => {
     copy(selectedAddress)
   }
 
+  React.useEffect(() => {
+    isSingle &&
+      multipleAddressesUsed &&
+      isShowingMultipleAddressInfo &&
+      openModal(strings.singleOrMultiple, <Modal />, modalHeight)
+  }, [isShowingMultipleAddressInfo, isSingle, multipleAddressesUsed, openModal, strings.singleOrMultiple])
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Receive/useCases/DescribeSelectedAddressScreen.tsx]: nitpicky consistency no worries, I'm considering that this should be part of the useAddressStatus -> status should report this kind of things on top of my head.
```suggestion
  }, [isShowingMultipleAddressInfo, isSingle, isMultipleAddressesUsed, openModal, strings.singleOrMultiple])
```

File: apps/wallet-mobile/src/features/Receive/useCases/DescribeSelectedAddressScreen.tsx [hunk 3]
```diff
@@ -31,6 +42,13 @@ export const DescribeSelectedAddressScreen = () => {
     copy(selectedAddress)
   }
 
+  React.useEffect(() => {
+    isSingle &&
+      multipleAddressesUsed &&
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Receive/useCases/DescribeSelectedAddressScreen.tsx]:
```suggestion
      isMultipleAddressesUsed &&
```

File: apps/wallet-mobile/src/features/Receive/useCases/DescribeSelectedAddressScreen.tsx [hunk 4]
```diff
@@ -2,25 +2,36 @@ import {useFocusEffect} from '@react-navigation/native'
...
+import {QRs} from '../illustrations/QRs'
 
 export const DescribeSelectedAddressScreen = () => {
   const strings = useStrings()
   const {styles, colors} = useStyles()
   const navigate = useNavigateTo()
   const {selectedAddress} = useReceive()
+  const {isSingle, addressMode} = useAddressModeManager()
+  const addresses = useReceiveAddressesStatus(addressMode)
+  const multipleAddressesUsed = addresses.used.length > 1
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/Receive/useCases/DescribeSelectedAddressScreen.tsx]:
```suggestion
  const isMultipleAddressesUsed = addresses.used.length > 1
```

**Review by stackchain — APPROVED** [review]

**Pull request merged**